text/0021-versioning-releases.md

```diff
@@ -0,0 +1,45 @@
+# Releasing and Versioning of Openmrs SingleSpa Apps
+- Start Date: 2020/01/21
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/1
+
+## Definition
+OpenMRS: refers to the openmrs community.
+
+Micro-Frontends Squad: refers to the OpenMRS Community members responsible for creating and maintaining OpenMRS SPAs, under the OpenMRS Micro-Frontends Project.
+
+OpenMRS SPA: refers to an OpenMRS Micro-Frontend app. An example is openmrs-esm-patient-chart. See https://github.com/openmrs/openmrs-esm-patient-chart
+
+CDN: Content Distribution Network such as jsdelivr. See https://www.jsdelivr.com/
+
+MAJOR Version: version where there are incompatible API or configuration API changes, or major UI changes that are enabled by default
+
+MINOR Version: version when you add functionality in a backwards compatible manner
+
+PATCH Version: version when you make backwards compatible bug fixes
+
+## Decision
+OpenMRS SPA applications should have releases that are versioned so that each implementor/distribution can decide when to upgrade to a certain version, containing features that the implementor/distribution needs. 
+
+The releases should have a maintenance branch for bug fixes and patches so that implementors running a particular version of an app can benefit from recent bug fixes. 
+
+I propose that we have maintenance release branches for both Major and Minor releases, as the motivation for this is to allow implementors/distributions control on when to start using a certain feature, that could be in a certain minor or major release.
+
+An implementor or distribution that requires a certain maintenance branch, either for a minor or major release is free to create it, by branching from the release-tagged commit.
+
+The OpenMRS Microfrontend Squad will not maintain release branches. Implementers or organizations who wish to maintain release branches may do so. They 
+should be granted write access to the GitHub repo hosting the ESM of interest.
```

Comment:
It might be worth mentioning that development should always occur on the master branch with selected changes _within master_ backported to release branches as needed. Making changes directly to a release branch and forwardporting those changes to master is discouraged and should only happen – if ever – in extreme circumstances.

Reply:
I completely agree!

Reply:
Agreed